src/Levenshtein.php

```diff
@@ -6,13 +6,29 @@ class Levenshtein
 {
     public static function distance(string $string1, string $string2, int $insertionCost = 1, $replacementCost = 1, $deletionCost = 1): int
     {
-        $string1 = mb_convert_encoding($string1, 'ASCII', 'utf8');
-        $string2 = mb_convert_encoding($string2, 'ASCII', 'utf8');
+        $map = [];
+        $string1 = self::utf8_to_extended_ascii($string1, $map);
+        $string2 = self::utf8_to_extended_ascii($string2, $map);
 
-        if (false === $string1 || false === $string2) {
-            throw new \InvalidArgumentException('Both, string1 and string2 have to be valid utf-8 strings.');
+        return levenshtein($string1, $string2, $insertionCost, $replacementCost, $deletionCost);
+    }
+
+    private static function utf8_to_extended_ascii($str, &$map)
+    {
+        // find all multibyte characters (cf. utf-8 encoding specs)
+        $matches = [];
+        if (!preg_match_all('/[\xC0-\xF7][\x80-\xBF]+/', $str, $matches)) {
+            return $str;
+        } // plain ascii string
+
+        // update the encoding map with the characters not already met
+        foreach ($matches[0] as $mbc) {
+            if (!isset($map[$mbc])) {
+                $map[$mbc] = \chr(128 + \count($map));
```

Comment:
We should probably ensure that `\count($map)` is lower than 128 here, or otherwise throw an exception.

```suggestion
                if (\count($map) >= 128) {
                    throw new \InvalidArgumentException('Strings with more than 128 individual unicode characters are not supported.');
                }
                $map[$mbc] = \chr(128 + \count($map));
```